kotsadm/pkg/handlers/login.go

```diff
@@ -178,13 +182,21 @@ func OIDCLoginCallback(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		if err := kotsadmdex.ResetDexState(state); err != nil {
+		if err := kotsadmdex.ResetDexState(r.Context(), namespace, state); err != nil {
 			logger.Error(errors.Wrap(err, "failed to reset dex state"))
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
 
-		ctx := oidc.ClientContext(r.Context(), http.DefaultClient)
+		httpClient, err := identity.HTTPClient(r.Context(), namespace)
+		if err != nil {
+			err = errors.Wrap(err, "failed to get identity http client")
+			logger.Error(err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		ctx := oidc.ClientContext(r.Context(), httpClient)
```

Comment:
will this work for existing clusters?

Reply:
ya, the cert just wont exist. like using the default client